week4/MAKEME.md

```diff
@@ -0,0 +1,72 @@
+# Homework React Week 4
+
+## **Todo list**
+
+1. Practice the concepts
+2. React exercises
+3. Code along
+4. PROJECT: Hack Your Weather IIII
+
+## **1. Practice the concepts**
+
+Let's finish the online exercises. Start from `Review Using Props with Stateless Functional Components` until the last exercise `Render React on the Server with renderToString`:
+
+- [FreeCodeCamp: Exercise 21-48](https://www.freecodecamp.org/learn/front-end-libraries/react/)
+
+## **2. React exercises**
+
+**No exercises this week**
+
+## **3. Code along**
+
+In the following tutorial you'll learn how to
+
+- [Build a Movie Database](https://www.youtube.com/playlist?list=PL_kAgwZgMfWyZ6m8fDwdiwEarr_g6nFxz)
+
+## **4. PROJECT: Hack Your Weather IV**
+
+> Make sure you're building on last week's codebase.
+
+This week we'll finish our application by integrating 2 things: `client-side routing` with [React-Router-Dom](https://reacttraining.com/react-router/web/guides/quick-start) and `charting` our data (in either hourly or five day forecasts).
+
+### 1. Setup
+
+1. Install the following packages: [React Router Dom](https://www.npmjs.com/package/react-router-dom) and [Recharts](https://www.npmjs.com/package/recharts)
+
+```
+npm install react-router-dom recharts
+```
+
+### 2. Requirements
+
+At the end of the week the user interface of your application will look mostly the same, but with an additional page:
+
+![Week 4 Wireframe](../assets/project/week4.png)
+
+Here are the requirements your project needs to fulfill:
+
+- Divide your page into 2 different routes (`/` and `/:cityId`), using `react-router-dom`
+- Allow users to click on the searched city to open the city page
+- The city page will be on a different route identified by the city id (make use of the `<Link>` component given by `react-router-dom`)
+- On the city page there will be several things:
+  - Two tabs, one for the hourly forecast and the other for a 5 day forecast
```

Comment:
i just realised the hourly forecast is now a paid api :/
so what we can do now is on clicking the city will show a page with only 5 day forecast (in charts) and not hourly forecast

Reply:
Ah that's too bad. Then we just keep it to the 5 day forecast.